src/reverselock.h

```diff
@@ -13,7 +13,7 @@ class reverse_lock
 {
 public:
 
-    explicit reverse_lock(Lock& lock) : lock(lock) {
+    explicit reverse_lock(Lock& _lock) : lock(_lock) {
```

Comment:
Thanks. Though, it seems I still don't get identical binaries. Could you try to create a pull with a single commit and combine #8655 + #8808 with all the changes that don't change the binary? Then we could evaluate the remaining (problematic) hunks separate from those.